temporalcli/commands.workflow.go

```diff
@@ -53,8 +70,68 @@ func (*TemporalWorkflowQueryCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
 
-func (*TemporalWorkflowResetCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowResetCommand) run(cctx *CommandContext, _ []string) error {
+	if c.Type.Value == "" && c.EventId <= 0 {
+		return errors.New("must specify either valid event id or reset type")
+	}
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	conn, err := c.Parent.ClientOptions.dialGRPC(cctx)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	wfsvc := workflowservice.NewWorkflowServiceClient(conn)
+	resetBaseRunID := c.RunId
+	eventID := int64(c.EventId)
+	if c.Type.Value != "" {
+		resetBaseRunID, eventID, err = getResetEventIDByType(cctx, c.Type.Value, c.Parent.Namespace, c.WorkflowId, c.RunId, wfsvc)
+		if err != nil {
+			return fmt.Errorf("getting reset event ID by type failed: %w", err)
+		}
+	}
+	username := "<unknown-user>"
+	if u, err := user.Current(); err != nil && u.Username != "" {
+		username = u.Username
+	}
```

Comment:
do not we have a function to establish username?

Reply:
One does not exist and I didn't extract these three lines into one, no

Reply:
you have exactly same code in defaultReason. Extaract and use